src/mito2/src/read/projection.rs

```diff
@@ -203,21 +227,159 @@ enum BatchIndex {
     Field(usize),
 }
 
+/// Gets a vector with repeated values from specific cache or creates a new one.
+fn repeated_vector_with_cache(
+    data_type: &ConcreteDataType,
+    value: &Value,
+    num_rows: usize,
+    cache_manager: &CacheManager,
+) -> common_recordbatch::error::Result<VectorRef> {
+    if let Some(vector) = cache_manager.get_repeated_vector(value) {
+        // Tries to get the vector from cache manager. If the vector doesn't
+        // have enough length, creates a new one.
+        match vector.len().cmp(&num_rows) {
+            Ordering::Less => (),
+            Ordering::Equal => return Ok(vector),
+            Ordering::Greater => return Ok(vector.slice(0, num_rows)),
+        }
+    }
+
+    // Creates a new one.
+    let vector = new_repeated_vector(data_type, value, num_rows)?;
+    // Updates cache.
+    cache_manager.put_repeated_vector(value.clone(), vector.clone());
```

Comment:
If a vector is excessively large, it will occupy memory for an extended period without being released. Therefore, when dealing with such a sizable vector, should we refrain from caching it?

Reply:
In this circumstance, the length of a batch is `<= batch_size` so there won't be a very large vector.

Reply:
I can also add a check here.